sklearn/tests/test_kernel_approximation.py

```diff
@@ -305,32 +308,38 @@ def test_input_validation(csr_container):
     RBFSampler().fit(X).transform(X)
 
 
-def test_nystroem_approximation():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_nystroem_approximation(array_namespace, device, dtype_name):
```

Comment:
I would rather not change the existing tests that check the behavior with regular numpy array when array API dispatch is disabled.

Instead add a new test that compares the results of using the estimator on array API inputs with array API dispatch enabled (and also check the type and device of fitted attributes) as we do for other array API estimators (e.g. `test_pca_array_api_compliance`) with non-default hyper-parameter values.

Note that you should also add `"array_api_support": True` to the `_more_tags` method of this estimator so that our common tests pick it up and run standard array API compliance tests with the default parameters.

Also use `array_api` somewhere in the test name so that it's easy to run all array API related test of scikit-learn by doing `pytest -k array_api sklearn`.